File: Scripts/results/generated_code_results/Flutter/bak_code/Social Networking/SocialApp/OnboardingScreen.dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
class OnboardingScreen extends StatelessWidget {
@override
Widget build(BuildContext context) {
SystemChrome.setSystemUIOverlayStyle(SystemUiOverlayStyle.light);
reasonml
复制
return Scaffold(
  body: Stack(
    children: [
      Image.asset(
        'assets/images/template.png',
        fit: BoxFit.cover,
        width: double.infinity,
        height: double.infinity,
      ),
      Container(
        decoration: BoxDecoration(
          gradient: LinearGradient(
            begin: Alignment.topCenter,
            end: Alignment.bottomCenter,
            colors: [Colors.transparent, Colors.black.withOpacity(0.7)],
          ),
        ),
      ),
      SafeArea(
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Padding(
              padding: const EdgeInsets.all(16.0),
              child: Row(
                children: [
                  Icon(Icons.circle, color: Colors.white, size: 24),
                  SizedBox(width: 8),
                  Text(
                    'heact',
                    style: TextStyle(
                      color: Colors.white,
                      fontSize: 24,
                      fontWeight: FontWeight.bold,
                    ),
                  ),
                ],
              ),
            ),
            Spacer(),
            Padding(
              padding: const EdgeInsets.all(16.0),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(
                    'Effortlessly Discover\nYour Ideal Match',
                    style: TextStyle(
                      color: Colors.white,
                      fontSize: 32,
                      fontWeight: FontWeight.bold,
                    ),
                  ),
                  SizedBox(height: 16),
                  Text(
                    'Experience a seamless journey uncovering the ideal match that aligns with your preferences and aspirations',
                    style: TextStyle(
                      color: Colors.white70,
                      fontSize: 16,
                    ),
                  ),
                  SizedBox(height: 32),
                  ElevatedButton(
                    onPressed: () {
                      Navigator.pushReplacementNamed(context, 'Homepage');
                    },
                    child: Row(
                      mainAxisAlignment: MainAxisAlignment.center,
                      children: [
                        Icon(Icons.apple),
                        SizedBox(width: 8),
                        Text('Continue with Apple'),
                      ],
                    ),
                    style: ElevatedButton.styleFrom(
                      backgroundColor: Colors.white,
                      foregroundColor: Colors.black,
                      minimumSize: Size(double.infinity, 50),
                      shape: RoundedRectangleBorder(
                        borderRadius: BorderRadius.circular(25),
                      ),
                    ),
                  ),
                  SizedBox(height: 16),
                  ElevatedButton(
                    onPressed: () {
                      Navigator.pushReplacementNamed(context, 'Homepage');
                    },
                    child: Row(
                      mainAxisAlignment: MainAxisAlignment.center,
                      children: [
                        Icon(Icons.facebook),
                        SizedBox(width: 8),
                        Text('Continue with Facebook'),
                      ],
                    ),
                    style: ElevatedButton.styleFrom(
                      backgroundColor: Colors.blue,
                      foregroundColor: Colors.white,
                      minimumSize: Size(double.infinity, 50),
                      shape: RoundedRectangleBorder(
                        borderRadius: BorderRadius.circular(25),
                      ),
                    ),
                  ),
                  SizedBox(height: 16),
                  ElevatedButton(
                    onPressed: () {
                      Navigator.pushReplacementNamed(context, 'Homepage');
                    },
                    child: Text('Use Phone Number'),
                    style: ElevatedButton.styleFrom(
                      backgroundColor: Colors.transparent,
                      foregroundColor: Colors.white,
                      minimumSize: Size(double.infinity, 50),
                      shape: RoundedRectangleBorder(
                        borderRadius: BorderRadius.circular(25),
                        side: BorderSide(color: Colors.white),
                      ),
                    ),
                  ),
                  SizedBox(height: 32),
                ],
              ),
            ),
          ],
        ),
      ),
    ],
  ),
);
}
}

File: Scripts/results/generated_code_results/RN_gpt/Social Networking/SocialApp/OnboardingScreen.js
import React from 'react';
import { View, Text, Button, StatusBar, StyleSheet, ImageBackground } from 'react-native';

const OnboardingScreen = ({ navigation }) => {
  return (
    <ImageBackground
      source={require('../assets/snack-icon.png')}
      style={styles.background}
    >
      <StatusBar barStyle="light-content" />
      <View style={styles.container}>
        <Text style={styles.timeText}>9:41</Text>
        <Text style={styles.logo}>heact</Text>
        <Text style={styles.title}>Effortlessly Discover Your Ideal Match</Text>
        <Text style={styles.subtitle}>
          Experience a seamless journey uncovering the ideal match that aligns with your preferences and aspirations
        </Text>

        <Button
          title="Continue with Apple"
          onPress={() => navigation.navigate('Homepage')}
          color="black"
        />
        <Button
          title="Continue with Facebook"
          onPress={() => navigation.navigate('Homepage')}
          color="#4267B2"
        />
        <Button
          title="Use Phone Number"
          onPress={() => navigation.navigate('Homepage')}
          color="black"
        />
      </View>
    </ImageBackground>
  );
};

const styles = StyleSheet.create({
  background: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  container: {
    alignItems: 'center',
    paddingHorizontal: 20,
  },
  timeText: {
    position: 'absolute',
    top: 10,
    left: 15,
    color: 'white',
    fontSize: 12,
  },
  logo: {
    marginVertical: 20,
    fontSize: 32,
    fontWeight: 'bold',
    color: 'white',
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    textAlign: 'center',
    color: 'white',
    marginVertical: 10,
  },
  subtitle: {
    fontSize: 16,
    textAlign: 'center',
    color: 'white',
    marginBottom: 20,
  },
});

export default OnboardingScreen;